aten/src/ATen/native/Pow.cpp

```diff
@@ -77,6 +77,70 @@ Tensor pow(Scalar base, const Tensor& exp) {
   return native::pow_out(result, base, exp);
 }
 
+Tensor& float_power_out(Tensor& result, const Tensor& base, const Tensor& exp) {
+  auto dtype = (at::isComplexType(base.scalar_type()) || at::isComplexType(exp.scalar_type())) ?
+                at::kComplexDouble : at::kDouble;
+  TORCH_CHECK(result.scalar_type() == dtype,
+              "output type ", result.scalar_type(), "is not the desired output type ", dtype);
+
+  auto iter = TensorIterator::binary_op(result, base.to(dtype), exp.to(dtype));
+  pow_tensor_tensor_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor& float_power_out(Tensor& result, const Tensor& base, Scalar exp) {
+  // Numpy compatibility check:
+  TORCH_CHECK(!(isIntegralType(base.scalar_type(), true) &&
+              exp.isIntegral(true) && exp.toLong() < 0),
+              "Integers to negative integer powers are not allowed.");
+
+  auto dtype = (at::isComplexType(base.scalar_type()) || exp.isComplex()) ? at::kComplexDouble : at::kDouble;
+  TORCH_CHECK(result.scalar_type() == dtype,
+              "output type ", result.scalar_type(), "is not the desired output type ", dtype);
+
+  if (exp.isComplex() && (exp.toComplexDouble() == 0.0) ) {
```

Comment:
These are neat but I think just calling the previous function should be OK. If we do keep them then we should use the `resize_output` function to resize.

Reply:
Progressing in the separate PR #46830